lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_send_app_size_metrics.rb

```diff
@@ -0,0 +1,212 @@
+require_relative '../../helper/app_size_metrics_helper'
+
+module Fastlane
+  module Actions
+    class AndroidSendAppSizeMetricsAction < Action
+      def self.run(params)
+        # Check input parameters
+        api_url = URI(params[:api_url])
+        api_token = params[:api_token]
+        if (api_token.nil? || api_token.empty?) && !api_url.is_a?(URI::File)
+          UI.user_error!('An API token is required when using an `api_url` with a scheme other than `file://`')
+        end
+
+        # Build the payload base
+        metrics_helper = Fastlane::Helper::AppSizeMetricsHelper.new(
+          Platform: 'Android',
+          'App Name': params[:app_name],
+          'App Version': params[:app_version_name],
+          'Version Code': params[:app_version_code],
+          'Product Flavor': params[:product_flavor],
+          'Build Type': params[:build_type],
+          Source: params[:source]
+        )
+        metrics_helper.add_metric(name: 'AAB File Size', value: File.size(params[:aab_path]))
+
+        # Add device-specific 'splits' metrics to the payload if a `:include_split_sizes` is enabled
+        if params[:include_split_sizes]
+          check_bundletool_installed!
+          apkanalyzer_bin = params[:apkanalyzer_binary] || find_apkanalyzer_binary!
+          UI.message("[App Size Metrics] Generating the various APK splits from #{params[:aab_path]}...")
+          Dir.mktmpdir('release-toolkit-android-app-size-metrics') do |tmp_dir|
+            Action.sh('bundletool', 'build-apks', '--bundle', params[:aab_path], '--output-format', 'DIRECTORY', '--output', tmp_dir)
+            apks = Dir.glob('splits/*.apk', base: tmp_dir).map { |f| File.join(tmp_dir, f) }
+            UI.message("[App Size Metrics] Generated #{apks.length} APKs.")
+
+            apks.each do |apk|
+              UI.message("[App Size Metrics] Computing file and download size of #{File.basename(apk)}...")
+              split_name = File.basename(apk, '.apk')
+              file_size = Action.sh(apkanalyzer_bin, 'apk', 'file-size', apk, print_command: false, print_command_output: false).chomp.to_i
+              download_size = Action.sh(apkanalyzer_bin, 'apk', 'download-size', apk, print_command: false, print_command_output: false).chomp.to_i
+              metrics_helper.add_metric(name: 'APK File Size', value: file_size, metadata: { split: split_name })
+              metrics_helper.add_metric(name: 'Download Size', value: download_size, metadata: { split: split_name })
+            end
```

Comment:
ℹ️  Note for the reviewers:
 - This code has since been extracted in a separate method during https://github.com/wordpress-mobile/release-toolkit/pull/365, to make the code cleaner, less "massive" and more readable
 - The names of the metrics (`APK File Size` and `Download Size` here, but same for all others used all over in `*_send_app_size_metrics`) have since been DRYed into constants in https://github.com/wordpress-mobile/release-toolkit/pull/365 as well